docs/string_grader.md

```diff
@@ -1,66 +1,236 @@
+
+```pycon
+>>> grader = StringGrader(
+...     answers='cat',
+...     strip=False
+... )
+>>> grader(None, 'cat') == {'grade_decimal': 1, 'msg': '', 'ok': True}
+True
+>>> grader(None, ' cat') == {'grade_decimal': 0, 'msg': '', 'ok': False}
+True
+
+```
+
+This will accept `cat`, but will reject answers with leading or trailing spaces. By default, `strip=True`.
+
+Similarly, if a student uses two (or more) spaces instead of one, that rarely changes the meaning. By default, we convert consecutive spaces into a single space before comparison (this applies to both the instructor-provided answer and the student-provided response). If you want to disable this behavior, you can set `clean_spaces=False` as follows.
+
+```pycon
+>>> grader = StringGrader(
+...     answers='two  spaces',
+...     clean_spaces=False
+... )
+>>> grader(None, 'two  spaces') == {'grade_decimal': 1, 'msg': '', 'ok': True}
+True
+>>> grader(None, 'two spaces') == {'grade_decimal': 0, 'msg': '', 'ok': False}
+True
+
+```
+
+Here, the answer is `two  spaces`, complete with two spaces. A students' answer of `two spaces` (with a single space) would be graded incorrect.
+
+Finally, you may have a situation where spaces are completely irrelevant (e.g., when grading a mathematical expression). To instruct the grader to completely ignore all spaces, set `strip_all=True`.
+
+```pycon
+>>> grader = StringGrader(
+...     answers='(12)(34)',
+...     strip_all=True
+... )
+>>> grader(None, '(1 2) (3 4)') == {'grade_decimal': 1, 'msg': '', 'ok': True}
+True
+
+```
+
+This grader will accept `(1 2) (3 4)`, ignoring all spaces in the provided answer.
+
+
 ## Case Sensitive
 
 To perform case-insensitive grading, pass in the appropriate flag as follows.
 
-```python
-grader = StringGrader(
-    answers='Cat',
-    case_sensitive=False
-)
+```pycon
+>>> grader = StringGrader(
+...     answers='Cat',
+...     case_sensitive=False
+... )
+>>> grader(None, 'Cat') == {'grade_decimal': 1, 'msg': '', 'ok': True}
+True
+>>> grader(None, 'cat') == {'grade_decimal': 1, 'msg': '', 'ok': True}
+True
+>>> grader(None, 'CAT') == {'grade_decimal': 1, 'msg': '', 'ok': True}
+True
+
 ```
 
 This will accept `Cat`, `cat` and `CAT`. By default, `case_sensitive=True`.
 
 
-## Stripping Input
+## Accepting anything (within reason!)
 
-Leading or trailing spaces in an answer rarely change the meaning of the answer. Hence, by default, we strip all leading and trailing spaces from the student input and author-specified answers before comparison. If you want to keep those spaces around for the comparison, you need to disable strip.
+Sometimes you may just want to accept anything that a student provides (possibly subject to conditions). To do this, to set the `accept_any` flag, which will literally accept anything that is entered into the textbox.
```

Comment:
Is it worth adding something about the motivation? "...perhaps for a free-response question". Feel free to reject.

Reply:
So added.